cg/meta/orders/api.py

```diff
@@ -11,47 +11,148 @@
+        FastqOrderSubmitter,
+    ),
+    OrderType.FLUFFY: (
+        OrderLimsService,
+        ValidatePoolOrderService,
+        StorePoolOrderService,
+        PoolOrderSubmitter,
+    ),
+    OrderType.METAGENOME: (
+        OrderLimsService,
+        ValidateMetagenomeOrderService,
+        StoreMetagenomeOrderService,
+        MetagenomeOrderSubmitter,
+    ),
+    OrderType.MICROSALT: (
+        OrderLimsService,
+        ValidateMicrobialOrderService,
+        StoreMicrobialOrderService,
+        MicrobialOrderSubmitter,
+    ),
+    OrderType.MIP_DNA: (
+        OrderLimsService,
+        ValidateGenericOrderService,
+        StoreGenericOrderService,
+        GenericOrderSubmitter,
+    ),
+    OrderType.MIP_RNA: (
+        OrderLimsService,
+        ValidateGenericOrderService,
+        StoreGenericOrderService,
+        GenericOrderSubmitter,
+    ),
+    OrderType.RML: (
+        OrderLimsService,
+        ValidatePoolOrderService,
+        StorePoolOrderService,
+        PoolOrderSubmitter,
+    ),
+    OrderType.RNAFUSION: (
+        OrderLimsService,
+        ValidateGenericOrderService,
+        StoreGenericOrderService,
+        GenericOrderSubmitter,
+    ),
+    OrderType.SARS_COV_2: (
+        OrderLimsService,
+        ValidateMicrobialOrderService,
+        StoreMicrobialOrderService,
+        MicrobialOrderSubmitter,
+    ),
+    OrderType.TOMTE: (
+        OrderLimsService,
+        ValidateGenericOrderService,
+        StoreGenericOrderService,
+        GenericOrderSubmitter,
+    ),
+}
 
-def _get_submit_handler(project: OrderType, lims: LimsAPI, status: Store) -> Submitter:
-    """Factory Method"""
-
-    submitters = {
-        OrderType.BALSAMIC: BalsamicSubmitter,
-        OrderType.BALSAMIC_QC: BalsamicQCSubmitter,
-        OrderType.BALSAMIC_UMI: BalsamicUmiSubmitter,
-        OrderType.FASTQ: FastqSubmitter,
-        OrderType.FLUFFY: FluffySubmitter,
-        OrderType.METAGENOME: MetagenomeSubmitter,
-        OrderType.MICROSALT: MicrosaltSubmitter,
-        OrderType.MIP_DNA: MipDnaSubmitter,
-        OrderType.MIP_RNA: MipRnaSubmitter,
-        OrderType.RML: RmlSubmitter,
-        OrderType.RNAFUSION: RnafusionSubmitter,
-        OrderType.SARS_COV_2: SarsCov2Submitter,
-        OrderType.TOMTE: TomteSubmitter,
-    }
-    if project in submitters:
-        return submitters[project](lims=lims, status=status)
+
+class OrderSubmitterFactory:
+    def __init__(self, status_db: Store, lims_api: LimsAPI):
+        self.status_db = status_db
+        self.lims_api = lims_api
+
+    def create_order_submitter(self, order_type: OrderType) -> OrderSubmitter:
+        service_classes = order_service_mapping.get(order_type)
+
+        if not service_classes:
+            raise ValueError(f"No services found for order type: {order_type}")
+
+        lims_service, validate_service_class, store_service_class, submitter_class = service_classes
+
+        lims_service: OrderLimsService = lims_service(self.lims_api)
```

Comment:
so regarding my previous comment, would this be possible for all order types?
```suggestion
        lims_service = OrderLimsService(self.lims_api)
```